servicepulse/install-servicepulse-in-iis.md

```diff
@@ -3,28 +3,92 @@ title: Install ServicePulse in IIS
 
 Once all the ServicePulse files are successfully extracted you can configure a new IIS web site whose physical path points to the location where files have been extracted.
 
-### Changing ServicePulse port
+When using IIS to host ServicePulse the ServicePulse.Host service is not used.  To remove the service uninstall ServicePulse from Add/Remove programs
+
+### Advanced Configuration
+
+ServicePulse relies on the ServiceControl REST API.  It is possible to add a reverse proxy to the web site using  the Microsoft [Application Request Routing](http://www.iis.net/downloads/microsoft/application-request-routing) IIS extension.
+This is useful if you which to lock down access to ServicePulse or if wish to expose the web site over a single port.
+
+Note: Please review the Security section below prior to carrying out the installation
+
+Installation Steps:
+
+1. Install the IIS ARR 3 extension (http://www.iis.net/downloads/microsoft/application-request-routing)
+- Go to the root folder for the Web site you created in the basic configuration
+- Create a new subdirectory called `api`
+- Edit `config.js` and change the `serviceControlUrl` value from `http://localhost:33333/api` to `/api`
+- Open the IIS management tool
+- Select the api sub folder from within IIS management tool
+- Click the `URL Rewrite`
+- Add a new URL Rewrite Rule
+- Choose `Reverse Proxy` from the list of rule templates
+- Enter `localhost:33333/api` into the inbound field and leave SSL offload enabled then click OK to add the rule.
+- The website should now answer on /api as though you were directly accessing the ServiceControl.  
+
+The procedure above should result in a web.config file in the newly created /api folder similar to this:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <system.webServer>
+        <rewrite>
+            <rules>
+                <rule name="ReverseProxyInboundRule1" stopProcessing="true">
+                    <match url="(.*)" />
+                    <action type="Rewrite" url="http://localhost:33333/api/{R:1}" />
+                </rule>
+            </rules>
+        </rewrite>
+    </system.webServer>
+</configuration>
+
+```
+### Upgrades
+
+When ServicePulse is hosted in IIS the upgrade process is as follows:
+
+1. Go to the root directory of the IIS web site,
+- View the contents of `config.js` and record the current value of `serviceControlUrl`
+- Remove all files and folders in the root of the IIS Web Site except the `api` folder if it exists
+- Install the new version of ServicePulse using the standard instructions
+- Extract the files from the ServicePulse.Host.exe using the following commandline, replacing `<rescordedvalue>` with the value from the `config.js` and `<webroot`> with the path to the root directory of the IIS website`
+```
+ServicePulse.Host.exe` --extract --serviceControlUrl="<rescordedvalue>" --outPath="<webroot>"
+```
+- Optionally uninstall ServicePulse to remove the unneeded Windows Service.
```

Comment:
assumed knowledge - I'll can add the text via Add/Removes programs if you want me to be more explicit that it's just a standard uninstall